spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreOperations.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.firestore;
+
+import java.util.List;
+
+import com.google.protobuf.Empty;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * An interface of operations that can be done with Cloud Firestore.
+ *
+ * @author Dmitry Solomakha
+ */
+public interface FirestoreOperations {
```

Comment:
Should we name it `FirestoreReactiveOperations`? Or is this going to be the only template implementation?

Reply:
Are `findById` , `save`, and `saveAll` coming later or in this PR?

Reply:
`save` are coming in this PR, the rest are coming later

Reply:
renamed